ote/src/clj/ote/services/pre_notices/operator.clj

```diff
@@ -16,14 +16,31 @@
 
 (defqueries "ote/services/pre_notices/regions.sql")
 
+(defn- parse-attachments
+  [attachments db user]
+  (let [saved-attachments (mapv
+                            #(first (specql/fetch db ::transit/pre-notice-attachment
+                                           (specql/columns ::transit/pre-notice-attachment)
+                                                  (op/and
+                                                  {::transit/attachment-file-name (::transit/attachment-file-name %)
+                                                   ::modification/created-by (get-in user [:user :id])})))
+                            attachments)]
+    saved-attachments))
+
 (defn get-operator-pre-notice [db user id]
   "Get singular operator pre-notice by id"
   (http/no-cache-transit-response
-    (first (specql/fetch
+    (let [pre-notice (first (specql/fetch
              db ::transit/pre-notice
              (specql/columns ::transit/pre-notice)
```

Comment:
Could you just join the attachments and let specql handle it? There are no other collections being fetched.